src/wasm/wasm-emscripten.cpp

```diff
@@ -757,29 +756,38 @@ void AsmConstWalker::visitCall(Call* curr) {
                 << ".\nThis might be caused by aggressive compiler "
                    "transformations. Consider using EM_JS instead.";
       }
-    } else if (auto* value = arg->dynCast<Binary>()) {
-      // In the dynamic linking case the address of the string constant
-      // is the result of adding its offset to __memory_base.
-      // In this case are only looking for the offset with the data segment so
-      // the RHS of the addition is just what we want.
-      assert(value->op == AddInt32);
-      arg = value->right;
-    } else {
-      if (!value) {
-        Fatal() << "Unexpected arg0 type (" << getExpressionName(arg)
-                << ") in call to: " << importName;
+      continue;
+    }
+
+    if (auto* setlocal = arg->dynCast<LocalSet>()) {
+      // The argument may be a local.tee, in which case we take first child
+      // which is the value being copied into the local.
+      if (setlocal->isTee()) {
+        arg = setlocal->value;
+        continue;
       }
     }
+
+    if (auto* bin = arg->dynCast<Binary>()) {
+      if (bin->op == AddInt32) {
+        // In the dynamic linking case the address of the string constant
+        // is the result of adding its offset to __memory_base.
+        // In this case are only looking for the offset from __memory_base
+        // the RHS of the addition is just what we want.
+        arg = bin->right;
+        continue;
+      }
+    }
+
+    Fatal() << "Unexpected arg0 type (" << getExpressionName(arg)
+            << ") in call to: " << importName;
   }
 
   auto* value = arg->cast<Const>();
-  auto code = codeForConstAddr(wasm, segmentOffsets, value);
-  auto& asmConst = createAsmConst(code, sig, importName);
+  int32_t address = value->value.geti32();
```

Comment:
Is there any reason `codeForConstAddr ` changed from taking a `Const*` to an `int32_t`? It seems like this was a partial change that didn't make it in to the final PR, so the effect was that we just outlined the `value->value.geti32()` bit.